packages/server/tests/acceptance/ws/estimateGas.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// external resources
+import { expect } from 'chai';
+import { Contract, ethers, JsonRpcProvider, WebSocketProvider } from 'ethers';
+import { AliasAccount } from '../../clients/servicesClient';
+import WebSocket from 'ws';
+import basicContractJson from '../../contracts/Basic.json';
+
+describe('@release @web-socket eth_estimateGas', async function () {
+  // @ts-ignore
+  const { servicesNode, relay } = global;
+
+  const RELAY_URL = `${process.env.RELAY_ENDPOINT}`;
+  const WS_RELAY_URL = `${process.env.WS_RELAY_URL}`;
+  const BASIC_CONTRACT_PING_CALL_DATA = '0x5c36b186';
+  const PING_CALL_ESTIMATED_GAS = '0x6122';
+
+  let accounts: AliasAccount[] = [],
+    basicContract: Contract,
+    currentPrice: number,
+    expectedGas: number,
+    gasPriceDeviation: number,
+    provider: JsonRpcProvider,
+    requestId: string,
+    wsProvider: WebSocketProvider,
+    webSocket: WebSocket;
+
+  before(async () => {
+    accounts[0] = await servicesNode.createAliasAccount(100, relay.provider, requestId);
+    basicContract = await servicesNode.deployContract(basicContractJson);
+    wsProvider = await new ethers.WebSocketProvider(WS_RELAY_URL);
+    provider = new ethers.JsonRpcProvider(RELAY_URL);
+
+    webSocket = new WebSocket(WS_RELAY_URL);
+    currentPrice = await relay.gasPrice(requestId);
+    expectedGas = parseInt(PING_CALL_ESTIMATED_GAS, 16);
+
+    gasPriceDeviation = parseFloat(expectedGas.toString() ?? '0.2');
```

Comment:
Maybe add an explaination on this logic?

Reply:
Done.